src/timezones/types.jl

```diff
@@ -222,6 +224,57 @@ function ZonedDateTime(local_dt::DateTime, tz::VariableTimeZone, is_dst::Bool; f
 end
 
 doc"""
+`ZonedDateTime(tz::VariableTimeZone, is_dst::Bool; from_utc::Bool=false) -> ZonedDateTime`
+
+Get a `ZonedDateTime` object in the timezone specified, for the current time.
+
+equivalent to `ZonedDateTime(now(Dates.UTC), tz, is_dst; from_utc=true)`
+"""
+ZonedDateTime(local_dt::DateTime, is_dst::Bool) = ZonedDateTime(now(Dates.UTC), tz, is_dst; from_utc=true)
+
+doc"""
+`ZonedDateTime(local_dt::DateTime, is_dst::Bool; from_utc::Bool=false) -> ZonedDateTime`
+
+Get a `ZonedDateTime` object for the time specified, using your local time zone.
+
+equivalent to `ZonedDateTime(local_dt, localzone(), is_dst; from_utc=from_utc)`
+"""
+ZonedDateTime(local_dt::DateTime, is_dst::Bool; from_utc::Bool=false) = ZonedDateTime(local_dt, localzone(), is_dst; from_utc=from_utc)
+
+doc"""
+`ZonedDateTime(tz::VariableTimeZone, occurrence::Integer=0) -> ZonedDateTime`
+
+Get a ZonedDateTime object in the timezone specified, for the current time (as provided by your system).
+
+equivalent to `ZonedDateTime(now(Dates.UTC), tz, occurrence; from_utc=true)`
+"""
+ZonedDateTime(tz::VariableTimeZone, occurrence::Integer=0) = ZonedDateTime(now(Dates.UTC), tz, occurrence; from_utc=true)
+
+doc"""
+`ZonedDateTime(local_dt::DateTime, is_dst::Bool; from_utc::Bool=false) -> ZonedDateTime`
+
+Get a `ZonedDateTime` object for the time specified, using your local time zone
+
+equivalent to `ZonedDateTime(local_dt, localzone(), occurrence; from_utc=from_utc)`
+"""
+ZonedDateTime(local_dt::DateTime, occurrence::Integer=0; from_utc::Bool=false) = ZonedDateTime(local_dt, localzone(), occurrence; from_utc=from_utc)
+
+doc"""
+`ZonedDateTime() -> ZonedDateTime`
+
+`ZonedDateTime(occurrence::Integer) -> ZonedDateTime`
+
+ZonedDateTime() will give you a ZonedDateTime for the current time in your local time zone.
+
+equivalent to `ZonedDateTime(now(Dates.UTC), localzone(); from_utc=true)` if occurrence is not set or
+
+equivalent to `ZonedDateTime(now(Dates.UTC), localzone(), occurrence; from_utc=true)` if it is
+"""
+ZonedDateTime(occurrence::Integer=0) = ZonedDateTime(now(Dates.UTC), localzone(); from_utc=true)
```

Comment:
This should probably be implemented as `now(localzone())`. This does lose millisecond precision though. If you really want millisecond precision:

```julia
function ZonedDateTime(occurrence::Integer=0)
    utc = unix2datetime(time())
    ZonedDateTime(utc, localzone(); from_utc=true)
end
```
I would like to avoid using the `now(Dates.UTC)` method as I'll probably push for its removal from Base.